pyqmc/coord.py

```diff
@@ -111,29 +106,37 @@ class PeriodicElectron:
     """
 
     def __init__(self, epos, lattice_vectors, dist, wrap=None):
+        assert len(epos.shape) <= 2, "PeriodicElectron is for single electron only"
         self.configs = epos
         self.lvec = lattice_vectors
         self.wrap = wrap if wrap is not None else np.zeros_like(epos)
         self.dist = dist
 
 
 class PeriodicConfigs:
-    def __init__(self, configs, lattice_vectors, wrap=None):
+    def __init__(self, configs, lattice_vectors, wrap=None, dist=None):
         configs, wrap_ = pbc.enforce_pbc(lattice_vectors, configs)
         self.configs = configs
         self.wrap = wrap_
         if wrap is not None:
             self.wrap += wrap
         self.lvecs = lattice_vectors
-        self.dist = distance.MinimalImageDistance(lattice_vectors)
+        self.dist = dist or distance.MinimalImageDistance(lattice_vectors)
```

Comment:
Please use if dist is None or similar. 

See for example: https://stackoverflow.com/questions/3914667/false-or-none-vs-none-or-false